src/ansys/dpf/core/helpers/streamlines.py

```diff
@@ -0,0 +1,182 @@
+"""Streamlines computation specific helpers.
+"""
+
+import numpy as np
+import warnings
+
+from ansys.dpf.core.common import locations
+from ansys.dpf.core.fields_container import FieldsContainer
+from ansys.dpf.core.helpers.utils import _sort_supported_kwargs
+
+
+class _PvFieldsContainerBase:
+    def __init__(self, data):
+        """Instantiate Streamline
+        from pyvista.PolyData object.
+        This construction is only
+        intended to be used internally.
+
+        Parameters
+        ----------
+        pv_data_set: pyvista.PolyData
+        """
+        try:
+            import pyvista as pv
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "To use streamlines capabilities, please install pyvista "
+                "with :\n pip install pyvista>=0.24.0"
+            )
+        self._pv_data_set = None
+        self._streamlines_fc = None
+        if isinstance(data, pv.PolyData):
+            self._pv_data_set = data
+        elif isinstance(FieldsContainer):
+            self._streamlines_fc = data
+        else:
+            raise AttributeError(
+                "streamlines must be a pyvista.PolyData or a dpf.FieldsContainer instance."
+            )
+
+    def _pv_data_set_to_fc(self):
+        """Convert pyvista.PolyData into FieldsContainer."""
+        raise Exception("Not implemented yet")
+
+    def _fc_to_pv_data_set(self):
+        """Convert pyvista.PolyData into FieldsContainer."""
+        raise Exception("Not implemented yet")
+
+    def _as_pyvista_data_set(self):
+        if self._pv_data_set is None:
+            self._pv_data_set = self._fc_to_pv_data_set()
+        return self._pv_data_set
+
+    def _as_fields_container(self):
+        """Returns a FieldsContainer representing the streamlines
+        related objects.
+
+        Returns
+        -------
+        streamlines_fields_container : FieldsContainer
+            FieldsContainer containing the streamlines data array,
+            and a MeshedRegion as support.
+
+        """
+        # once implemented, this method will become public
+        if self._streamlines_fc is None:
+            self._streamlines_fc = self._pv_data_set_to_fc()
+        return self._streamlines_fc
+
+
+class Streamlines(_PvFieldsContainerBase):
+    """Class to define the Streamlines object
+    scripting with `ansys-dpf-core`.
+
+    """
+
+    def __init__(self, data):
+        super().__init__(data=data)
+
+
+class StreamlinesSource(_PvFieldsContainerBase):
+    """Class to define the StreamlinesSource
+    object scripting with `ansys-dpf-core`.
+
+    """
+
+    def __init__(self, data):
+        super().__init__(data=data)
+
+
+def compute_streamlines(meshed_region, field, **kwargs):
+    """Compute the streamlines for a given mesh and velocity
```

Comment:
will we be able to put a sphere object here?

Reply:
@cbellot000 for now this is not possible to pass a "source" of any kind (even pyvista object) to compute the streamline. Computing streamlines from a source using "compute_streamlines" from a pyvista object source should be possible with some (small) changes. As far as we have that available, we could use the "helpers.streamlines.StreamlinesSource" object to handle the Geometry objects (however I am seeing that currently, only the "Line" and "Point" exists there). 

Is it needed to have the "compute_streamline" from a specific source changes in that PR ? Or can that be done later ?

Reply:
@anslpa @cbellot000 I guess the idea is to add an optional "source" argument in the future which will take in a geometry object

Reply:
@PProfizi yes, or create a "compute_streamlines_from_source" method as PyVista did